app/src/main/java/com/android/sample/App.kt

```diff
@@ -1,6 +1,48 @@
 package com.android.sample
 
 import android.app.Application
+import com.android.sample.helper.NotificationHelper
+import com.android.sample.model.activity.Activity
 import dagger.hilt.android.HiltAndroidApp
+import java.time.LocalTime
+import java.util.concurrent.TimeUnit
 
-@HiltAndroidApp class App : Application()
+@HiltAndroidApp
+class App : Application() {
+  lateinit var notificationHelper: NotificationHelper
+
+  companion object {
+    private lateinit var instance: App
+
+    fun getInstance(): App {
+      return instance
+    }
+  }
+
+  override fun onCreate() {
+    super.onCreate()
+    instance = this
+    notificationHelper = NotificationHelper(this)
+  }
+
+  fun scheduleNotification(activity: Activity, isCreator: Boolean) {
```

Comment:
I have tried a different approach too but had a hard time to launch everything. In most tutorials I have seen, they do it this way.

Reply:
My concern was essentially that it could make e2e testing very hard for me for this part. If things are not implemented fully according to the MVVM structure.

Let's try it this way, and just hope that it does not trouble Hilt injection. If any issues occur, I'll take care of it... 👍

Reply:
Oh I did not think of that. Normally it should be ok bc in the tutorial I've seen, they also have both implementation in app and hilt declaration. Hopefully there is no problem later...